tests/integration/managedclusteraddon_test.go

```diff
@@ -0,0 +1,146 @@
+			managedcluster1 := clusterv1.ManagedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster1",
+				},
+			}
+			managedcluster2 := clusterv1.ManagedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster2",
+				},
+			}
+			ns1 := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster1",
+				},
+			}
+			ns2 := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster2",
+				},
+			}
+			By("creating MirrorPeer. Also, this should automatically create ManagedClusterAddOn", func() {
+				err := k8sClient.Create(context.TODO(), &managedcluster1, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &managedcluster2, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &ns1, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &ns2, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				err = k8sClient.Create(context.TODO(), newMirrorPeer, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+			})
+			By("polling for the created ManagedClusterAddOn", func() {
+				Eventually(func() error {
+					err := k8sClient.Get(context.TODO(), mcAddOn1LookupKey, &mcAddOn1)
+					if err != nil {
+						return err
+					}
+					return fmt.Errorf("Waiting on ManagedClusterAddOn %s/%s to get created", mcAddOn1LookupKey.Namespace, mcAddOn1LookupKey.Name)
+				}, 60*time.Second, 2*time.Second).ShouldNot(HaveOccurred())
+
+				Eventually(func() error {
+					err := k8sClient.Get(context.TODO(), mcAddOn2LookupKey, &mcAddOn2)
+					if err != nil {
+						return err
+					}
+					return fmt.Errorf("Waiting on ManagedClusterAddOn %s/%s to get created", mcAddOn2LookupKey.Namespace, mcAddOn2LookupKey.Name)
+				}, 60*time.Second, 2*time.Second).ShouldNot(HaveOccurred())
+			})
+			By("updating ManagedClusterAddOn", func() {
+				mcAddOn1.Spec.InstallNamespace = "new-test-namespace"
+				err := k8sClient.Update(context.TODO(), &mcAddOn1, &client.UpdateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				err = k8sClient.Delete(context.TODO(), &mcAddOn1, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Delete(context.TODO(), &mcAddOn2, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Delete(context.TODO(), newMirrorPeer, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Delete(context.TODO(), &ns1, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Delete(context.TODO(), &ns2, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Delete(context.TODO(), &managedcluster1, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Delete(context.TODO(), &managedcluster2, &client.DeleteOptions{})
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
all these should be done in `After()` block, it'll be called automatically after the test.